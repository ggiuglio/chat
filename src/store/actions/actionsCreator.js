import {
  LOAD_MESSAGES,
  WRITE_MESSAGE
} from './actionsTypes.js'
import axios from 'axios';

export const loadMessages = () => {
  return (dispatch, getState) => {
    const token = getState().myToken;
    return axios.get(`https://chatty.kubernetes.doodle-test.com/api/chatty/v1.0/?token=${token}`)
      .then((res) => {
        dispatch({
          type: LOAD_MESSAGES,
          messages: res.data
        });
      })
      .catch((err) => {
        console.log(err)
      });
  }
}

export const postMessage = (text) => {
  return (dispatch, getState) => {
    const author = getState().myName;
    const token = getState().myToken;
    return axios.post(
      'https://chatty.kubernetes.doodle-test.com/api/chatty/v1.0',
      { "message": text, "author": author },
      {
        headers: {
          'content-type': 'application/json',
          'token': token
        }
      }
    ).then((res) => {

      dispatch({
        type: WRITE_MESSAGE,
        message: res.data
      });
    })
      .catch((err) => {
        console.log(err)
      });
  }
}