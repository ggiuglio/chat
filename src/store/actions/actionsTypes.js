export const LOAD_MESSAGES = "LOAD_MESSAGES";
export const WRITE_MESSAGE = "WRITE_MESSAGE";