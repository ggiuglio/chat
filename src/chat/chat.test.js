import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { loadMessagesMock } from '../store/actions/actionsCreator';
import ReactTestUtils from 'react-dom/test-utils';


import Chat from './chat';
const { act } = renderer;


const mockStore = configureStore([]);

describe('Chat container', () => {
  let store;
  let component;
  const stateMessages = [
    {
      "author": "Tom",
      "_id": 0,
      "isMine": false,
      "message": "Hi this is my first message",
      "date": "1/19/1970",
    },
    {
      "author": "Lisa",
      "_id": 1,
      "isMine": false,
      "message": "How is the project going?",
      "date": "1/19/1970",
    },
    {
      "author": "Tom",
      "_id": 2,
      "isMine": false,
      "message": "It proceeds fine?",
      "date": "1/19/1970",
    },
    {
      "author": "Giulio",
      "_id": 3,
      "isMine": true,
      "message": "Hi, can I have access to the shared folder?",
      "date": "1/19/1970",
    },
  ];

  beforeEach(() => {
    store = mockStore({
      messages: stateMessages
    });
    store.dispatch = jest.fn();
    act(() => {
      component = renderer.create(
        <Provider store={store}>
          <Chat />
        </Provider>
      );
    });
  });

  it('should dispatch loadMessagesMock action', () => {
    expect(store.dispatch).toHaveBeenCalledTimes(1)
  });

  it('should render message list', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

});