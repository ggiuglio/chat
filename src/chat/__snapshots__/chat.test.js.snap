// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat container should render message list 1`] = `
<div
  className="sc-gzVnrw ejgSRj"
>
  <div
    className="sc-htoDjs gcGsnJ"
  >
    <div
      className="sc-bdVaJa bVzWCR"
    >
      <div
        className="sc-bwzfXH bGLOCH"
      >
        Tom
      </div>
      <div
        className="sc-htpNat SYTas"
      >
        Hi this is my first message
      </div>
      <div
        className="sc-bxivhb dMhSOm"
      >
        1/19/1970
      </div>
    </div>
    <div
      className="sc-bdVaJa bVzWCR"
    >
      <div
        className="sc-bwzfXH bGLOCH"
      >
        Lisa
      </div>
      <div
        className="sc-htpNat SYTas"
      >
        How is the project going?
      </div>
      <div
        className="sc-bxivhb dMhSOm"
      >
        1/19/1970
      </div>
    </div>
    <div
      className="sc-bdVaJa bVzWCR"
    >
      <div
        className="sc-bwzfXH bGLOCH"
      >
        Tom
      </div>
      <div
        className="sc-htpNat SYTas"
      >
        It proceeds fine?
      </div>
      <div
        className="sc-bxivhb dMhSOm"
      >
        1/19/1970
      </div>
    </div>
    <div
      className="sc-bdVaJa iOpiwT"
    >
      
      <div
        className="sc-htpNat SYTas"
      >
        Hi, can I have access to the shared folder?
      </div>
      <div
        className="sc-bxivhb dMhSOm"
      >
        1/19/1970
      </div>
    </div>
  </div>
  <div
    className="sc-ifAKCX fRTgzm"
  >
    <textarea
      className="sc-EHOje hCBww"
      onChange={[Function]}
      value=""
    />
    <button
      className="sc-bZQynM kdMRmz"
      onClick={[Function]}
    >
      Send
    </button>
  </div>
</div>
`;
